### ISSUE TITLE
RA infinite redirect

### ISSUE THREAD

**Issue by stenington** [Iridescent-CM/technovation-app#2321]
**Describe the bug**
I don't know if this happens in production, but it has happened multiple times locally. When I try to log in as `ra@ra.com`, I sometimes get an infinite redirect error page.

```
Started GET "/regional_ambassador/location_details?return_to=%2Fregional_ambassador%2Fdashboard" for 127.0.0.1 at 2019-12-05 12:17:47 -0500
[SPOOFING IP] 200.94.199.219
Processing by RegionalAmbassador::LocationDetailsController#show as HTML
  Parameters: {"return_to"=>"/regional_ambassador/dashboard"}
  Account Load (0.5ms)  SELECT  "accounts".* FROM "accounts" WHERE "accounts"."deleted_at" IS NULL AND "accounts"."auth_token" = $1 LIMIT $2  [["auth_token", "LoMay2ShhLhAbuY5F7L1buHu"], ["LIMIT", 1]]
  SignupAttempt Load (0.3ms)  SELECT  "signup_attempts".* FROM "signup_attempts" WHERE "signup_attempts"."account_id" = $1 LIMIT $2  [["account_id", 6], ["LIMIT", 1]]
  RegionalAmbassadorProfile Load (0.3ms)  SELECT  "regional_ambassador_profiles".* FROM "regional_ambassador_profiles" WHERE "regional_ambassador_profiles"."account_id" = $1 LIMIT $2  [["account_id", 6], ["LIMIT", 1]]
Redirected to http://localhost:3000/regional_ambassador/profile#!location
Completed 302 Found in 57ms (ActiveRecord: 1.1ms)


Started GET "/regional_ambassador/profile" for 127.0.0.1 at 2019-12-05 12:17:47 -0500
[SPOOFING IP] 200.94.199.219
Processing by RegionalAmbassador::ProfilesController#show as HTML
  Account Load (1.1ms)  SELECT  "accounts".* FROM "accounts" WHERE "accounts"."deleted_at" IS NULL AND "accounts"."auth_token" = $1 LIMIT $2  [["auth_token", "LoMay2ShhLhAbuY5F7L1buHu"], ["LIMIT", 1]]
  SignupAttempt Load (1.3ms)  SELECT  "signup_attempts".* FROM "signup_attempts" WHERE "signup_attempts"."account_id" = $1 LIMIT $2  [["account_id", 6], ["LIMIT", 1]]
  RegionalAmbassadorProfile Load (0.5ms)  SELECT  "regional_ambassador_profiles".* FROM "regional_ambassador_profiles" WHERE "regional_ambassador_profiles"."account_id" = $1 LIMIT $2  [["account_id", 6], ["LIMIT", 1]]
Redirected to http://localhost:3000/regional_ambassador/location_details?return_to=%2Fregional_ambassador%2Fdashboard
Filter chain halted as :require_location rendered or redirected
Completed 302 Found in 14ms (ActiveRecord: 2.9ms)


Started GET "/regional_ambassador/location_details?return_to=%2Fregional_ambassador%2Fdashboard" for 127.0.0.1 at 2019-12-05 12:17:47 -0500
[SPOOFING IP] 200.94.199.219
Processing by RegionalAmbassador::LocationDetailsController#show as HTML
  Parameters: {"return_to"=>"/regional_ambassador/dashboard"}
  Account Load (0.5ms)  SELECT  "accounts".* FROM "accounts" WHERE "accounts"."deleted_at" IS NULL AND "accounts"."auth_token" = $1 LIMIT $2  [["auth_token", "LoMay2ShhLhAbuY5F7L1buHu"], ["LIMIT", 1]]
  SignupAttempt Load (0.4ms)  SELECT  "signup_attempts".* FROM "signup_attempts" WHERE "signup_attempts"."account_id" = $1 LIMIT $2  [["account_id", 6], ["LIMIT", 1]]
  RegionalAmbassadorProfile Load (0.3ms)  SELECT  "regional_ambassador_profiles".* FROM "regional_ambassador_profiles" WHERE "regional_ambassador_profiles"."account_id" = $1 LIMIT $2  [["account_id", 6], ["LIMIT", 1]]
Redirected to http://localhost:3000/regional_ambassador/profile#!location
Completed 302 Found in 44ms (ActiveRecord: 1.2ms)


Started GET "/regional_ambassador/profile" for 127.0.0.1 at 2019-12-05 12:17:47 -0500
[SPOOFING IP] 200.94.199.219
Processing by RegionalAmbassador::ProfilesController#show as HTML
  Account Load (0.8ms)  SELECT  "accounts".* FROM "accounts" WHERE "accounts"."deleted_at" IS NULL AND "accounts"."auth_token" = $1 LIMIT $2  [["auth_token", "LoMay2ShhLhAbuY5F7L1buHu"], ["LIMIT", 1]]
  SignupAttempt Load (0.4ms)  SELECT  "signup_attempts".* FROM "signup_attempts" WHERE "signup_attempts"."account_id" = $1 LIMIT $2  [["account_id", 6], ["LIMIT", 1]]
  RegionalAmbassadorProfile Load (0.4ms)  SELECT  "regional_ambassador_profiles".* FROM "regional_ambassador_profiles" WHERE "regional_ambassador_profiles"."account_id" = $1 LIMIT $2  [["account_id", 6], ["LIMIT", 1]]
Redirected to http://localhost:3000/regional_ambassador/location_details?return_to=%2Fregional_ambassador%2Fdashboard
Filter chain halted as :require_location rendered or redirected
Completed 302 Found in 8ms (ActiveRecord: 1.5ms)
```

**To Reproduce**

**Comment by rsgonzal** [comment]
Need to investigate whether this happens in production

**Comment by shaunxp20** [comment]
I ran into this issue locally as well, didn't have time to look at it though. 🙁 😐 🙂

**Comment by stenington** [comment]
Oh, I believe I have a fix for this, which slipped in to #2351: https://github.com/Iridescent-CM/technovation-app/pull/2351/files#diff-f7a25bc30a409fc77050703fc048ca8cR236

**Comment by shaunxp20** [comment]
> Oh, I believe I have a fix for this, which slipped in to #2351: https://github.com/Iridescent-CM/technovation-app/pull/2351/files#diff-f7a25bc30a409fc77050703fc048ca8cR236

Awesome! 💯 🙌 I'll try to check it out tomorrow!

**Comment by stenington** [comment]
The fix is in the DB seed data, so what I did was something like `bundle exec rails db:drop` then `bin/setup` I think? Then logging in as the admin worked fine.

**Comment by shaunxp20** [comment]
I verified the fix in #2351, way to go @stenington! 👍 🎉 I could this bug being really annoying/tricky to figure out! 😬

And fwiw, I just ran `rails db:setup` (the step from `bin/setup`) to rebuild/repopulate the database. 🤷‍♂ 👍 😄
